test/parallel/test-signal-handler.js

```diff
@@ -22,14 +22,14 @@ var i = 0;
 setInterval(function() {
   console.log('running process...' + ++i);
 
-  if (i == 5) {
+  if (i === 5) {
     process.kill(process.pid, 'SIGUSR1');
   }
 }, 1);
 
 // Test on condition where a watcher for SIGNAL
 // has been previously registered, and `process.listeners(SIGNAL).length === 1`
-process.on('SIGHUP', function() {});
+process.on('SIGHUP', function() { common.fail('should not run'); });
```

Comment:
@jasnell Done that way, the AssertionError looks like this:

```
AssertionError: null undefined null
    at process.exports.fail (/Users/trott/io.js/test/common.js:438:10)
    at emitNone (events.js:91:20)
    at process.emit (events.js:185:7)
    at Signal.wrap.onsignal (internal/process.js:199:44)
```

There are two things I don't like about that: 

* The stack trace does not show where the actual problem is. It does not mention the test file at all.
* `null undefined null`: What?!

In contrast, done the way it is here, it looks like this:

```
AssertionError: should not run
    at Object.exports.fail (/Users/trott/io.js/test/common.js:438:10)
    at process.<anonymous> (/Users/trott/io.js/test/parallel/test-signal-handler.js:32:42)
    at emitNone (events.js:91:20)
    at process.emit (events.js:185:7)
    at Signal.wrap.onsignal (internal/process.js:199:44)
```

The stack trace includes the name of the test file and the line that resulted in the failure.